Src/Particle/AMReX_ParticleTransformation.H

```diff
@@ -394,12 +394,55 @@ auto filterParticles (DstTile& dst, const SrcTile& src, Pred&& p,
     amrex::ParallelForRNG(np,
     [=] AMREX_GPU_DEVICE (int i, amrex::RandomEngine const& engine) noexcept
     {
-        p_mask[i] = particle_detail::call_f(p, src_data, i, engine);
+        p_mask[i] = particle_detail::call_f(p, src_data, src_start+i, engine);
```

Comment:
I believe it's correct, the `filterParticles` call loops over `mask[src_start+i]`  for `i=0,..,n-1`.

https://github.com/AMReX-Codes/amrex/pull/2185/files/039eb01c9bdc40bbedc0dfe5b6c4cb7d88f7ec53#diff-8e42446c3fbcc60f09f881dbf68c630fc1019e5481eb1c9e253710b948e8e3a3L319-L345

Reply:
This `ParallelForRNG` call goes over the full `np` in the src tile, so if `src_start` is not 0, doesn't that go out of bounds? Likewise, the `exclusive_sum` in the nested called to `filterParticles` goes over the full range of the mask, from 0 to np, so I think those values all need to be filled with 0 or 1.

You could make this so that both `mask` and `offsets` are size `n`, which in principle would be more efficient in memory and in terms of the number of times the predicate is applied, but I don't think you need to do that for this PR.

Reply:
I misread it, I thought both loops were over `n`, not `np`. Is there a compelling reason not to treat the `ParallelForRNG(np` as a typo that should actually be `ParallelForRNG(n`? If both `mask` and `offsets` are size `n`, then it's consistent with the comments above the function, and seems more logical to have a function that would have a `src_start` that's non-zero.